runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -619,28 +640,35 @@ private static void validateWorkerOptions(DataflowWorkerHarnessOptions options)
         StreamingDataflowWorker.class.getSimpleName());
   }
 
-  private static ChannelCache createChannelCache(
-      DataflowWorkerHarnessOptions workerOptions, ComputationConfig.Fetcher configFetcher) {
-    ChannelCache channelCache =
-        ChannelCache.create(
-            (currentFlowControlSettings, serviceAddress) -> {
-              // IsolationChannel will create and manage separate RPC channels to the same
-              // serviceAddress.
-              return IsolationChannel.create(
+  private static ChannelCache createChannelCacheForLocalWindmillService(
+      DataflowWorkerHarnessOptions workerOptions) {
+    return ChannelCache.create(
+        (currentFlowControlSettings, serviceAddress) -> {
+          // IsolationChannel will create and manage separate RPC channels to the same
+          // serviceAddress.
+          if (serviceAddress.getKind()
+              == WindmillServiceAddress.Kind.AUTHENTICATED_GCP_SERVICE_ADDRESS) {
+            return IsolationChannel.create(
+                () ->
+                    remoteChannel(
+                        serviceAddress,
+                        workerOptions.getWindmillServiceRpcChannelAliveTimeoutSec(),
+                        currentFlowControlSettings),
+                currentFlowControlSettings.getOnReadyThresholdBytes());
+          }
+          return workerOptions.getUseWindmillIsolatedChannels()
+              ? IsolationChannel.create(
```

Comment:
this seems the same as above, and the remote channel creation is always the same. Can we simplify as
Supplier<...> createChannel = () -> remoteChannel(....);
if ((serviceAddress.getKind()
              == WindmillServiceAddress.Kind.AUTHENTICATED_GCP_SERVICE_ADDRESS || workerOptions.getUseWindmillIsolatedChannels()) {
  return IsolationChannel.create(createChannel, currentFlowControlSettings.getOnReadThresholdBytes());
}
return createChannel();
}